packages/web3-eth-abi/src/types.ts

```diff
@@ -33,21 +33,23 @@ export type AbiParameter = {
 type FragmentTypes = 'constructor' | 'event' | 'function' | 'fallback';
 
 export type AbiBaseFragment = {
-	readonly type: FragmentTypes;
+	readonly type: string | FragmentTypes;
```

Comment:
Why must we add arbitrary `string`?

Reply:
If one uses Typescript but the ABI is declaring  without `as const` then the types for these attributes are not matched with the union values, so why we have to use a generic type for that cases.